ndcube/ndcube.py

```diff
@@ -607,11 +610,108 @@ def _fill_in_crop_nones(self, lower_corner, upper_corner, wcs, crop_by_values):
 
-    def _bounding_box_to_points(self, lower_corner_values, upper_corner_values, wcs):
+    def _bounding_box_to_points(self, lower_corner_values, upper_corner_values,
+                                axis_correlation_matrix):
         """
         Convert two corners of a bounding box to the points of all corners.
         """
-        return tuple(itertools.product(*zip(lower_corner_values, upper_corner_values)))
+        # Calculate which world axes share multiple pixel axes.
+        world_n_dep_dim = axis_correlation_matrix.sum(axis=1)
+        # If all world coordinates are independent, bounding box is simple.
+        max_dep_dim = world_n_dep_dim.max()
+        if max_dep_dim < 2:
+            return (tuple(lower_corner_values), tuple(upper_corner_values))
+        # If the number of world and pixel dims is the same, bounding box is also simple.
+        world_n_dim, pixel_n_dim = axis_correlation_matrix.shape
+        if world_n_dim == pixel_n_dim:
+            return tuple(itertools.product(*zip(lower_corner_values, upper_corner_values)))
+
+        # Otherwise we need calculate the corners more carefully.
+        # This must be done based on correlation matrix as not all
+        # world axes are independent.
+        # Start by generating array of world corner values assuming
+        # all coordinates are 1D. Strip units and add them back on later.
+        corner_units = [None] * len(lower_corner_values)
+        for i, (lcv, ucv) in enumerate(zip(lower_corner_values, upper_corner_values)):
+            if not isinstance(ucv, type(lcv)):
+                raise TypeError("Corresponding entries in lower and upper corner values "
+                                "must be of same type.")
+            if isinstance(lcv, u.Quantity):
+                corner_units[i] = lcv.unit
+                lower_corner_values[i] = lower_corner_values[i].value
+                upper_corner_values[i] = ucv.to_value(corner_units[i])
+        lower_corner_values = np.asarray(lower_corner_values)
+        upper_corner_values = np.asarray(upper_corner_values)
+        corners = np.stack([lower_corner_values, upper_corner_values])
+
+        # Next, calculate the sets of pixel axes upon which each world axis depends.
+        world_axes = np.arange(world_n_dim)
+        dep_pix_axes = np.array([set(np.arange(pixel_n_dim)[axis_correlation_matrix[j]])
+                                 for j in world_axes], dtype=object)
+
+        # Iterate through number of shared pixel axes, i, and create additional corners.
+        # Iterate from highest to lowest number of shared axes to reduce duplication.
+        # Do not include world axes sharing only 1 pixel axis, as we have already
+        # captured those in our initial definition of corners, above.
+        for i in range(max_dep_dim, 1, -1):
+            # Extract the world axes whose number of shared pixel axes is i.
+            world_axes_idx = np.where(world_n_dep_dim == i)[0]
```

Comment:
This case is still handled.  The above line of code (and the one below it) gets the world axis (axes) which correspond to `i` number of pixel axes.  At this point, the number of pixel axes is checked, but the specific set of pixel axes can be different or the same.

[The next line](https://github.com/sunpy/ndcube/pull/452/files#diff-929794bb532be1486907b360285433b432f8505ab7e7ff63eca3eab657f7d00dR657) then retrieves the set (sets) of pixel axes corresponding to each above world axis and stores them in `dep_pix_axes_i`.  At this stage, if there are two world axes that both share `i` pixel axes, but not the same set of pixel axes, the entries in `dep_pix_axes_i` will be different.  If both world axes do correspond to the same set of pixel axes, the entries in `dep_pix_axes_i` will be the same.

[The while loop](https://github.com/sunpy/ndcube/pull/452/files#diff-929794bb532be1486907b360285433b432f8505ab7e7ff63eca3eab657f7d00dR660) loops through each world axis and corresponding set of dependent pixel axes and builds the required corners.  But that loop also checks the sets of remaining dependent pixel axes in `dep_pix_axes_i`.  If any are the same as the current set, remaining sets are deleted from `dep_pix_axes_i` along with the corresponding world axis in `world_axes_i`.

This means corners are not duplicated by subsequent iterations of the `while` loop.  But it also means that if the set of dependent pixel axes of another world axis has the same length, but ISN'T the same set, it will not be deleted.  Hence its corners will still be calculated by the `while` loop.